src/Bot.ts

```diff
@@ -1,15 +1,17 @@
-import { Client, ClientOptions } from "discord.js";
+import { Client, GatewayIntentBits } from "discord.js";
 import * as dotenv from "dotenv";
+import ready from "./events/ready";
+import interactionCreate from "./events/interactionCreate";
 
 dotenv.config();
 
 const token = process.env.TOKEN;
 
-console.log("Bot is starting...");
-
 const client = new Client({
-  intents: [],
+  intents: [GatewayIntentBits.Guilds],
 });
-client.login(token);
 
-console.log(client);
+ready(client);
```

Comment:
I think globbing the events would be nicer.

```typescript
const eventFiles = await globPromise(
      `${__dirname}/../events/**/*{.ts,.js}`
    );
    eventFiles.forEach(async (filePath) => {
      const event: Event<keyof ClientEvents> = await this.importFile(filePath);
      this.on(event.event, event.run);
    });
```
Then inside `events` folder, you will have `ready.ts` which would look like this:
```typescript
export default new Event("ready", (client: Client) => {
        if(!client.user || !client.application) {
            return;
        }

        const commands = Commands.map(c => c.data);
        await client.application.commands.set(commands);

        console.log(`${client.user.tag} is online`);
});
```

Those requires a Event class
```typescript
export class Event<Key extends keyof ClientEvents> {
  constructor(
    public event: Key,
    public run: (...args: ClientEvents[Key]) => any
  ) {}
}
```

Reply:
> I think globbing the events would be nicer.
> 
> ```ts
> const eventFiles = await globPromise(
>       `${__dirname}/../events/**/*{.ts,.js}`
>     );
>     eventFiles.forEach(async (filePath) => {
>       const event: Event<keyof ClientEvents> = await this.importFile(filePath);
>       this.on(event.event, event.run);
>     });
> ```
> 
> Then inside `events` folder, you will have `ready.ts` which would look like this:
> 
> ```ts
> export default new Event("ready", (client: Client) => {
>         if(!client.user || !client.application) {
>             return;
>         }
> 
>         const commands = Commands.map(c => c.data);
>         await client.application.commands.set(commands);
> 
>         console.log(`${client.user.tag} is online`);
> });
> ```
> 
> Those requires a Event class
> 
> ```ts
> export class Event<Key extends keyof ClientEvents> {
>   constructor(
>     public event: Key,
>     public run: (...args: ClientEvents[Key]) => any
>   ) {}
> }
> ```

I did actually think of doing something like this (which is a similar approach to the commands), I didn't feel it was necessary since it's unlikely our use-case will need more than a couple of events and I wanted to avoid over-engineering. For the sake of consistency (and the obvious benefit if we actually needed several events) I'll add it. I think it's a great idea.